assets/js/lib/locations/triageLocation.js

```diff
@@ -0,0 +1,44 @@
+class TriageLocation {
+    constructor(dataRow) {
+        this.type = "TriageLocation";
+        this.title = dataRow[0];
+        this.latitude = parseFloat(dataRow[2]);
+        this.longitude = parseFloat(dataRow[3]);
+        this.url = dataRow[4];
+        if ((isNaN(this.latitude) || isNaN(this.longitude)) && Boolean(this.url)) {
+            try {
+                var url_location = this.url.split('@')[1].split('z')[0].split(',');
+                this.latitude = parseFloat(url_location[0]);
+                this.longitude = parseFloat(url_location[1]);
+
+            } catch (e) {
+                console.log(`Error in parsing URL for ${this.type} for title: ${this.title}`);
+                console.log(e);
+            }
+        }
+
+        this.nodal_officer = dataRow[6];
+
+        this.nodal_officer_contact = [];
+
+        if (dataRow[7] !== undefined) {
+            var contacts_info = dataRow[7].split(',')
+
+            for (var idx = 0; idx < contacts_info.length; idx++) {
```

Comment:
use [`.map`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map) here as well.

Also, watch out for bad variable names. You call the variable `this.nodal_officer_contact` which implies singular while it actually holds an array of multiple contacts. Use plural form for the variable name as well to improve it's readability.

Reply:
Please update all the variable names accordingly also use .map wherever necessary.